lib/bothan/dashboards.rb

```diff
@@ -37,6 +37,7 @@ def self.registered(app)
         @dashboard = Dashboard.find_by(slug: params[:slug])
 
         dashboard = params[:dashboard]
+        dashboard[:metrics].delete_if {|k,v| !v.has_key?("name")}
```

Comment:
The code here and on line 71 looks very similar. Could it be refactored into a separate method and called from both places, to avoid drift?